src/data/finset/pointwise.lean

```diff
@@ -888,11 +888,40 @@ coe_injective.no_zero_smul_divisors _ coe_zero coe_smul_finset
 
 end instances
 
-@[to_additive] lemma pairwise_disjoint_smul_iff [decidable_eq α] [left_cancel_semigroup α]
-  {s : set α} {t : finset α} :
+section left_cancel_semigroup
+variables [left_cancel_semigroup α] [decidable_eq α] (s t : finset α) (a : α)
+
+@[to_additive] lemma pairwise_disjoint_smul_iff {s : set α} {t : finset α} :
   s.pairwise_disjoint (• t) ↔ ((s : set α) ×ˢ (t : set α) : set (α × α)).inj_on (λ p, p.1 * p.2) :=
 by simp_rw [←pairwise_disjoint_coe, coe_smul_finset, set.pairwise_disjoint_smul_iff]
 
+@[to_additive] lemma card_singleton_mul : ({a} * t).card = t.card :=
```

Comment:
Should this one be `simp`?

Reply:
No because `finset.singleton_mul` already is marked simp. But maybe it shouldn't 🤔

Reply:
I've usually found it more awkward than helpful for that to be a simp lemma, personally

Reply:
"that" being `card_singleton_mul` or `singleton_mul`?

Reply:
`singleton_mul`